### PR TITLE
Add excludeDatabases for TPC-DS catalogs

### LINKED ISSUE #2759: [Subtask] Add excludeDatabases for TPC-DS catalogs
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

It's dangerous to expose huge scales catalog to users when setup a public playground

image

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/2538

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Close #2759

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalogSuite.scala [hunk 1]
```diff
@@ -48,6 +50,20 @@ class TPCDSCatalogSuite extends KyuubiFunSuite {
...
+    Seq("TINY,sf10" -> 2, "sf1 , " -> 1, "none" -> 0).foreach {
+      case (excludeDatabases, excludeNum) =>
+        val sparkConf = new SparkConf().setMaster("local[*]")
+          .set("spark.ui.enabled", "false")
+          .set("spark.sql.catalogImplementation", "in-memory")
+          .set("spark.sql.catalog.tpcds", classOf[TPCDSCatalog].getName)
+          .set("spark.sql.catalog.tpcds.excludeDatabases", excludeDatabases)
+        withSparkSession(SparkSession.builder.config(sparkConf).getOrCreate()) { spark =>
+          assert(spark.sql(s"SHOW DATABASES IN tpcds").count == 11 - excludeNum)
+        }
```
Comment (jiaoqingbo) [inline extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalogSuite.scala:63]: It is more accurate to check whether the result database does not contain excludeDatabases

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalog.scala [hunk 2]
```diff
@@ -21,27 +21,35 @@ import java.util
...
   override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
-    this.options = options
     this._name = name
+    this.options = options
+    this.excludeDatabases = options.get("excludeDatabases")
+      .split(",").map(_.toLowerCase.trim).filter(_.nonEmpty)
+    (this.excludeDatabases diff TPCDSSchemaUtils.DATABASES).foreach { unknownDb =>
+      logWarning(s"Ignore unknown database $unknownDb in excluding list")
+    }
   }
```
[the remark this answers is not in the document]
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalog.scala:57]: Make sense, updated

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalogSuite.scala (hunk 1 above)
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalogSuite.scala]: updated

**Review by jiaoqingbo — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**